src/issuance.ts

```diff
@@ -47,8 +52,57 @@ export async function getIssuerUrl(issuerName: string): Promise<string> {
     }
 }
 
-export interface TokenRequestProtocol {
-    serialize(): Uint8Array;
+// Send TokenRequest to Issuer (fetch w/POST).
+export async function sendTokenRequest(
+    tokReqBytes: Uint8Array,
+    issuerUrl: string,
+    headers?: Headers,
+): Promise<{ tokResBytes: Uint8Array }> {
+    headers ??= new Headers();
+    headers.append('Content-Type', MediaType.PRIVATE_TOKEN_REQUEST);
+    headers.append('Accept', MediaType.PRIVATE_TOKEN_RESPONSE);
+    const issuerResponse = await fetch(issuerUrl, {
+        method: 'POST',
+        headers,
+        body: tokReqBytes,
+    });
+
+    if (issuerResponse.status !== 200) {
+        const body = await issuerResponse.text();
+        throw new Error(`tokenRequest failed with code:${issuerResponse.status} response:${body}`);
```

Comment:
note for future: having a dedicated file with errors would allow some sort of matching down the line, and a more consistent information passed to the users.